creator-node/src/services/stateMachineManager/stateReconciliation/updateReplicaSet.jobProcessor.js

```diff
@@ -1,23 +1,550 @@
+      `ERROR issuing update replica set op: userId=${userId} wallet=${wallet} old replica set=[${primary},${secondary1},${secondary2}] | Error: ${e.toString()}`
+    )
+  }
+
+  // TODO: Return data about updated replica set or failure
   return {}
 }
+
+/**
+ * Logic to determine the new replica set. Used in reconfig op.
+ *
+ * The logic below is as follows:
+ * 1. Select the unhealthy replica set nodes size worth of healthy nodes to prepare for issuing reconfig
+ * 2. Depending the number and type of unhealthy nodes in `unhealthyReplicaSet`, issue reconfig depending on if the reconfig mode is enabled:
+ *  - if one secondary is unhealthy -> {primary: current primary, secondary1: the healthy secondary, secondary2: new healthy node}
+ *  - if two secondaries are unhealthy -> {primary: current primary, secondary1: new healthy node, secondary2: new healthy node}
+ *  - ** if one primary is unhealthy -> {primary: higher clock value of the two secondaries, secondary1: the healthy secondary, secondary2: new healthy node}
+ *  - ** if one primary and one secondary are unhealthy -> {primary: the healthy secondary, secondary1: new healthy node, secondary2: new healthy node}
+ *  - if entire replica set is unhealthy -> {primary: null, secondary1: null, secondary2: null, issueReconfig: false}
+ *
+ * ** - If in the case a primary is ever unhealthy, we do not want to pre-emptively issue a reconfig and cycle out the primary. See `peerSetManager` instance variable for more information.
+ *
+ * Also, there is the notion of `issueReconfig` flag. This value is used to determine whether or not to issue a reconfig based on the curretly enabled reconfig mode. See `RECONFIG_MODE` variable for more information.
+ *
+ * @param {Object} param
+ * @param {Object} param.logger the logger that can be filtered by jobName and jobId
+ * @param {string} param.primary current user's primary endpoint
+ * @param {string} param.secondary1 current user's first secondary endpoint
+ * @param {string} param.secondary2 current user's second secondary endpoint
+ * @param {string} param.wallet current user's wallet address
+ * @param {Set<string>} param.unhealthyReplicasSet a set of endpoints of unhealthy replica set nodes
+ * @param {string[]} param.healthyNodes array of healthy Content Node endpoints used for selecting new replica set
+ * @param {Object} param.replicaSetNodesToUserClockStatusesMap map of secondary endpoint strings to (map of user wallet strings to clock value of secondary for user)
+ * @param {string[]} param.enabledReconfigModes array of which reconfig modes are enabled
+ * @returns {Object}
+ * {
+ *  newPrimary: {string | null} the endpoint of the newly selected primary or null,
+ *  newSecondary1: {string | null} the endpoint of the newly selected secondary #1,
+ *  newSecondary2: {string | null} the endpoint of the newly selected secondary #2,
+ *  issueReconfig: {boolean} flag to issue reconfig or not
+ * }
+ */
+const determineNewReplicaSet = async ({
```

Comment:
this is 99% the same as snapback's [determineNewReplicaSet()](https://github.com/AudiusProject/audius-protocol/blob/master/creator-node/src/snapbackSM/snapbackSM.js#L606) -- just adapted for the logger changes. same with the rest of the functions in this class -- just refer to snapback for the original code